app/server/models/organisation.py

```diff
@@ -0,0 +1,124 @@
+from sqlalchemy import event, inspect
+from sqlalchemy.orm.query import Query
+from flask import g, current_app
+
+from server import db
+from server.exceptions import (
+    OrganisationNotProvidedException
+)
+from server.utils.blockchain_tasks import (
+    create_blockchain_wallet
+)
+from server.models.utils import ModelBase, OneOrgBase, ManyOrgBase, organisation_association_table
+
+@event.listens_for(Query, "before_compile", retval=True)
```

Comment:
is it okay for this to be in `organisation.py`? wasn't sure how this code works, didn't see it called directly anywhere, but seems relevant to organisation?

Reply:
i think this should probably be in utils or models.py. while it checks the org model, it's bound to ALL db queries via an event listener (https://docs.sqlalchemy.org/en/13/core/event.html)

Reply:
when does the code get run? on compilation do all event listeners anywhere in the code get run? like if i move it to utils, do i have to do something special to ensure it still happens as before? i'd prefer it not to be in models.py, eventually it'd be good if everything was moved out and that file didn't exist anymore

Reply:
>  eventually it'd be good if everything was moved out and that file didn't exist anymore

Oh yep good point. Utils would be an excellent place to put it. 

>  on compilation do all event listeners anywhere in the code get run? like if i move it to utils, do i have to do something special to ensure it still happens as before?

I'm 90% sure you can just shift it. I'll give it a shot once I've merged this PR